airbrake/notifier.py

```diff
@@ -83,6 +70,30 @@ def __init__(self, project_id=None, api_key=None, environment=None,
                 "by passing in the arguments explicitly."
             )
 
+        environment = config.get("environment",
+                                 os.getenv('AIRBRAKE_ENVIRONMENT'))
+
+        hostname = config.get("hostname", os.getenv('HOSTNAME') or
+                              socket.gethostname())
+
+        default_base_url = os.getenv('AIRBRAKE_BASE_URL',
+                                     self.AIRBRAKE_BASE_URL_DEFAULT.strip('/'))
+        base_url = config.get("base_url", default_base_url)
+
+        self.environment = str(environment)
+        self.base_url = str(base_url)
+        self.hostname = str(hostname)
+
+        self.component = config.get("component")
+        self.action = config.get("action")
+        self.app_version = config.get("app_version")
+        self.app_url = config.get("app_url")
+        self.user_agent = config.get("user_agent")
```

Comment:
Our schema docs say "The requesting browser’s full user-agent string.". So I think we'll be able to improve this in more framework-level integrations like airbrake-django. I think the reasonable default of `None` here is fine really. We have no clue what kind of software we're running inside here, and it only sounds relevant for web stuff.